streams/src/main/java/org/apache/kafka/streams/state/internals/CachingSessionStore.java

```diff
@@ -97,7 +100,7 @@ private void putAndMaybeForward(final ThreadCache.DirtyEntry entry, final Intern
                         binaryKey.get(),
                         newValueBytes,
                         sendOldValues ? oldValueBytes : null,
-                        entry.entry().context().timestamp());
+                        dslUsage ? bytesKey.window().end() : entry.entry().context().timestamp());
```

Comment:
Shall we reuse `entry.entry().context()` on L94 here, instead of calling it twice for `timestamp()`?